### PR TITLE
refactor: change payload hash to 32 byte array 

### PR DESCRIPTION
**Pull request by cgorenflo**

### CODE REVIEW

File: contracts/connection-router/src/state/payload_hex.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+pub fn deserialize<'de, D>(deserializer: D) -> Result<[u8; 32], D::Error>
+where
+    D: Deserializer<'de>,
+{
+    HexBinary::deserialize(deserializer)?
+        .to_array::<32>()
+        .map_err(|err| Error::custom(err))
+}
```
Comment (fish-sammy) [inline contracts/connection-router/src/state/payload_hex.rs:19]: Why not?
```suggestion
pub fn serialize<S, const N: usize>(value: &[u8; N], serializer: S) -> Result<S::Ok, S::Error>
where
    S: Serializer,
{
    HexBinary::from(value).serialize(serializer)
}

pub fn deserialize<'de, D, const N: usize>(deserializer: D) -> Result<[u8; N], D::Error>
where
    D: Deserializer<'de>,
{
    HexBinary::deserialize(deserializer)?
        .to_array::<N>()
        .map_err(|err| Error::custom(err))
}
```

Comment (fish-sammy) [reply contracts/connection-router/src/state/payload_hex.rs:19]: Also, I am not a fan of this mod name. It's really just serializing `[u8: N]` as hex string. Shouldn't have to be tied with payload_hash.

File: contracts/voting-verifier/src/events.rs [hunk 2]
```diff
@@ -145,7 +145,7 @@ impl TryFrom<Message> for TxEventConfirmation {
             destination_address: other.destination_address,
             destination_chain: other.destination_chain,
             source_address: other.source_address,
-            payload_hash: other.payload_hash,
+            payload_hash: HexBinary::from(other.payload_hash),
```
Comment (fish-sammy) [inline contracts/voting-verifier/src/events.rs:148]: Why is this event `TxEventConfirmation` not changed?

File: contracts/connection-router/src/state/payload_hex.rs (hunk 1 above)
Comment (cgorenflo) [reply contracts/connection-router/src/state/payload_hex.rs:19]: nice! I wasn't aware you could do `const N`. The reason I went with `payload_hash` is that I had it as `hex` first, but that collides with the std libs hex crate and I couldn't figure out how to resolve that naming conflict. I'm open for suggestions

File: contracts/voting-verifier/src/events.rs (hunk 2 above)
Comment (cgorenflo) [reply contracts/voting-verifier/src/events.rs:148]: Didn't want to broaden the scope of the PR and didn't check what exactly this is. Let me see if it makes sense to include it into this PR

File: contracts/connection-router/src/state.rs [hunk 3]
```diff
@@ -98,7 +96,7 @@ pub struct Message {
     pub destination_address: Address,
     /// for better user experience, the payload hash gets encoded into hex at the edges (input/output),
     /// but internally, we treat it as raw bytes to enforce it's format.
-    #[serde(with = "payload_hex")]
+    #[serde(with = "axelar_wasm_std::hex")]
```
Comment (fish-sammy) [inline contracts/connection-router/src/state.rs:99]: 👍

**Review by fish-sammy — APPROVED** [review]

**Pull request merged**